langchain/agents/tools/json.ts

```diff
@@ -0,0 +1,110 @@
+import jsonpointer from "jsonpointer";
+import { Tool, Toolkit } from "./base";
+
+export type Json =
+  | string
+  | number
+  | boolean
+  | null
+  | { [key: string]: Json }
+  | Json[];
+
+export type JsonObject = { [key: string]: Json };
+
+export class JsonSpec {
+  obj: JsonObject;
+
+  max_value_length = 4000;
```

Comment:
usually in JS we use camel case, so `maxValueLength`, unless we're trying to keep the API exactly the same as python